src/helpers.ts

```diff
@@ -144,20 +144,8 @@ export function toIterable<T>(i: any): any {
     }
     if (isPromiseLike(i)) {
         return {
-            [$A](): AsyncIterator<T> {
-                let finished: boolean;
-                return {
-                    next(): Promise<IteratorResult<T>> {
-                        if (finished) {
-                            return Promise.resolve({
-                                value: undefined,
-                                done: true,
-                            });
-                        }
-                        finished = true;
-                        return i.then((value: T) => ({value, done: false}));
-                    },
-                };
+            async *[$A]() {
+                yield await i;
```

Comment:
@vitaly-t Are you all good with switching to using generator functions like this? It can really cut down on the amount of code and complexity.